### PR TITLE
Repeating group summary2 table mode

### PR DESCRIPTION
**Pull request by Magnusrm**
## Description

Adds a table summary for repeating group.
Enables the use of Text, Number and Date components in Repeating group.

## Related Issue(s)

- closes #{issue number}

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [x] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Review by olemartinorg — APPROVED** [review]
Good stuff! :raised_hands: A few things to note:
- Automated tests for this should be added
- I guess the displayData stuff has been added so that rendering a Text component via `children` in a `RepeatingGroup` works - but won't that also show the text component in row edit mode? :thinking: I would think this already worked if setting `tableColumns[key].editInTable: true`. In that mode we wouldn't have to count on the `displayData` functionality so much, but it's nice to have regardless.

File: src/layout/RepeatingGroup/Summary2/RepeatingGroupTableSummary/RepeatingGroupTableSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            <Table.Row key={row?.uuid}>
+              {childNodes.map((node) => (
+                <DataCell
+                  node={node}
+                  key={node.id}
+                />
+              ))}
+              {!pdfModeActive && (
+                <Table.Cell
+                  align={'right'}
```
Comment (paal2707) [inline src/layout/RepeatingGroup/Summary2/RepeatingGroupTableSummary/RepeatingGroupTableSummary.tsx:97]: nitpick: trenger ikke curlybrackets

File: src/layout/RepeatingGroup/Summary2/RepeatingGroupTableSummary/RepeatingGroupTableSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                  className={tableClasses.buttonCell}
+                >
+                  {row?.items && row?.items?.length > 0 && (
+                    <EditButton
+                      componentNode={childNodes[0]}
+                      summaryComponentId=''
```
Comment (paal2707) [inline src/layout/RepeatingGroup/Summary2/RepeatingGroupTableSummary/RepeatingGroupTableSummary.tsx:103]: Hvorfor sendes tom streng inn her? kunne det vært håndtert i komponenten selv?

**Review by paal2707 — APPROVED** [review]
Ser bra ut! :)

File: src/layout/RepeatingGroup/Summary2/RepeatingGroupTableSummary/RepeatingGroupTableSummary.tsx (hunk 2 above)
Comment (Magnusrm) [reply src/layout/RepeatingGroup/Summary2/RepeatingGroupTableSummary/RepeatingGroupTableSummary.tsx:103]: Godt innspill! 👍

**Pull request merged**